### PR TITLE
[Feature] Improve `OBBject` Registry

### PR DESCRIPTION
**Pull request by hjoaquim**
1. **Why**? (1-3 sentences or a bullet point list):

    - Improve user experience by avoiding unnecessary confusion around the `results` command.

2. **What**? (1-3 sentences or a bullet point list):

    - `results` as a global command, available everywhere
    - help message on `results`
    - newly created `OBBjects` immediately available after creation
    - the obbject registry is now an instance of the session instead of a singleton
    - added flag to control number of obbject results to keep
    - added flag to control if the user wants to know when a new obbject result was added

3. **Impact** (1-2 sentences or a bullet point list):

    - NA / better UX

4. **Testing Done**:

    - Run a set of commands and access the results.

### CODE REVIEW

File: cli/openbb_cli/controllers/feature_flags_controller.py [hunk 1]
```diff
@@ -40,6 +39,8 @@ class FeatureFlagsController(BaseController):
         "language",
         "n_rows",
         "n_cols",
+        "obbject_registry_msg",
+        "obbject_registry_res",
```
Comment (montezdesousa) [inline cli/openbb_cli/controllers/feature_flags_controller.py:43]: Is this a typo?

```suggestion
        "obbject_msg",
        "obbject_res",
```

File: cli/openbb_cli/controllers/feature_flags_controller.py
Comment (montezdesousa) [inline cli/openbb_cli/controllers/feature_flags_controller.py:1]: Not related but we should rename this file/class to `settings_controller` to match the command name, becomes more intuitive.

**Comment by montezdesousa** [comment]
The OBBject message goes a bit unnoticed, I'd move it next to the remaining warnings about columns and --export. 

Screenshot 2024-05-06 at 11 42 36

**Review by montezdesousa — APPROVED** [review]

**Review by montezdesousa — COMMENTED** [review]
I'm getting `AttributeError: 'Namespace' object has no attribute 'columns'` when running

`/settings/ $ obbject_res -n 1`

File: cli/openbb_cli/controllers/feature_flags_controller.py
Comment (hjoaquim) [reply cli/openbb_cli/controllers/feature_flags_controller.py:1]: done here: https://github.com/OpenBB-finance/OpenBBTerminal/pull/6364/commits/5a9e18aac11800fbdd91db4f20e129609463622c

**Comment by hjoaquim** [comment]
> I'm getting `AttributeError: 'Namespace' object has no attribute 'columns'` when running
> 
> `/settings/ $ obbject_res -n 1`

typo fixed here: https://github.com/OpenBB-finance/OpenBBTerminal/pull/6364/commits/66a6ceac723c824ae97e7e64c3435eef17b05a85

**Comment by hjoaquim** [comment]
> The OBBject message goes a bit unnoticed, I'd move it next to the remaining warnings about columns and --export.
> Screenshot 2024-05-06 at 11 42 36

Addressed here: https://github.com/OpenBB-finance/OpenBBTerminal/pull/6364/commits/06184dee30811c7a10d4ab5204e1b791dfb04e42

**Pull request merged**